### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by ZuhRa-HashiMi**
CRUD_pic
ROLLBACK
Use database transactions.
Modify and delete data in SQL.
Prepare complex queries that answer analytical questions.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -33,3 +33,63 @@ WHERE NAME <> 'Gabumon';
...
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
```
Comment (omar-muhamad) [inline queries.sql:53]: - [ ] Nice job verifying that change persists after the commit, As per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Commit%20the%20transaction.-,Verify%20that%20change%20was%20made,-and%20persists%20after), Please verify that change was made before the commit too.

File: queries.sql [hunk 2]
```diff
@@ -33,3 +33,63 @@ WHERE NAME <> 'Gabumon';
...
+
+UPDATE animals
+SET species = 'pokemon';
+COMMIT;
+SELECT * FROM animals;
```
Comment (omar-muhamad) [inline queries.sql]: - [ ] This is not a required transaction, So please delete it.

File: queries.sql [hunk 3]
```diff
@@ -33,3 +33,63 @@ WHERE NAME <> 'Gabumon';
...
+COMMIT;
+
+SELECT count(*) FROM animals;
+SELECT count(*) FROM animals
+WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) AS average_weight FROM animals;
+SELECT * FROM animals
+ORDER BY escape_attempts DESC
+LIMIT 1;
```
Comment (omar-muhamad) [inline queries.sql]: - [ ] This query suppose to answer `Who escapes the most, neutered or not neutered animals?`, not who animal escapes the most? Please fix that.
[Hint] It's supposed to be like that:
```sql
SELECT neutered, MAX(escape_attempts) FROM animals
GROUP BY neutered;
```

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @ZuhRa-HashiMi,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] I can see that you use the Microverse README template, but please edit every section of your README file by removing the guiding sentences and filling them with the proper data about your project to make it more professional.

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by MaganaAsiati — APPROVED** [review]
# Status: Approved [✔️](https://emojipedia.org/check-mark/)
## Hi @ZuhRa-HashiMi    , I'm @MaganaAsiati , your reviewer for this PR  :wave: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


clap

## Highlights
- descriptive README file  :heavy_check_mark:👨🏻‍💻
- Your commits should start with a capital letter :heavy_check_mark:👨🏻‍💻


## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] [OPTIONAL] Please make sure to make your commit messages more descriptive in the future.
- [ ] [OPTIONAL]  kindly  Add a meaningful description in your pull request in the future which give more information about what you implement in this milestone :heavy_check_mark:👨🏻‍💻


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Cheers and Happy coding!👏👏👏

**Pull request merged**